addons/io/org.openhab.io.azureiothub/src/main/java/org/openhab/io/internal/azureiothub/CloudClient.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.internal.azureiothub;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.time.Instant;
+import java.util.function.Consumer;
+
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.items.events.ItemEventFactory;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.microsoft.azure.eventhubs.EventData;
+import com.microsoft.azure.eventhubs.EventHubClient;
+import com.microsoft.azure.eventhubs.PartitionReceiver;
+import com.microsoft.azure.sdk.iot.device.Message;
+import com.microsoft.azure.sdk.iot.service.Device;
+import com.microsoft.azure.servicebus.ServiceBusException;
+
+/**
+ *
+ * This class provides communication between openHAB and Azure IoT Hub.
+ * implementation is taken from https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-java-java-getstarted
+ *
+ * @author Niko Tanghe - Initial contribution
+ * @author Kai Kreuzer - Code cleanup
+ *
+ */
+
+public class CloudClient {
+
+    private final Logger logger = LoggerFactory.getLogger(CloudClient.class);
+
+    private String connectionstring;
+    private boolean commandEnabled;
+    private EventPublisher eventPublisher;
+
+    private AzureDevices azureDeviceStore;
+    private EventHubClient azureClient;
+    private Object lockobj = new Object();
+
+    /**
+     * Constructor of CloudClient
+     *
+     * @param connectionstring the connectionstring to the Azure IoT Hub
+     * @param eventPublisher
+     * @throws IOException
+     *
+     */
+    public CloudClient(String connectionstring, boolean commandEnabled, EventPublisher eventPublisher)
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.